x-pack/plugins/security_solution/public/management/pages/policy/view/tabs/policy_tabs.tsx

```diff
@@ -38,11 +53,30 @@ export const PolicyTabs = React.memo(() => {
   const isInHostIsolationExceptionsTab = usePolicyDetailsSelector(isOnHostIsolationExceptionsView);
   const policyId = usePolicyDetailsSelector(policyIdFromParams);
   const policyItem = usePolicyDetailsSelector(policyDetails);
+  const privileges = useUserPrivileges().endpointPrivileges;
+
+  const allPolicyHostIsolationExceptionsListRequest = useFetchHostIsolationExceptionsList({
+    page: 1,
+    perPage: 100,
+    policies: [policyId, 'all'],
+    // only enable if privileges are not loading and can not isolate a host
+    enabled: !privileges.loading && !privileges.canIsolateHost,
+  });
 
-  const tabs = useMemo(
-    () => [
-      {
-        id: 'settings',
+  const canSeeHostIsolationExceptions =
+    privileges.canIsolateHost || allPolicyHostIsolationExceptionsListRequest.data?.total !== 0;
+
+  // move the use out of this route if they can't access it
+  useEffect(() => {
+    if (isInHostIsolationExceptionsTab && !canSeeHostIsolationExceptions) {
```

Comment:
There is no loader while this is fetching for results, I think we should display a loader while we are checking if we can display the host isolation exceptions tab or not

Reply:
[`cd71ba9` (#121861)](https://github.com/elastic/kibana/pull/121861/commits/cd71ba95247d9e304fa3b4d4708c21d42816fca6)

Reply:
I'm thinking about it and I'm wondering if this check should be at a top level (at policy_details.tsx). There is already a loader in that component showed while we are fetching the policy. Wondering if would be better to use that loader instead of creating a new one in a deeper component

Reply:
I rather don't do it there. It means we'll be delaying the load of the page for all cases and not only for host isolation exceptions specific case.